packages/terra-functional-testing/src/reporters/spec-reporter/merge-results.js

```diff
@@ -0,0 +1,118 @@
+const fs = require('fs');
+const path = require('path');
+const cleanResults = require('./clean-results');
+const getOutputDir = require('./get-output-dir');
+
+/**
+ * Retrieves spec result files from a directory.
+ * @param {string} directory - The directory of the spec result files.
+ * @returns {array} - An array of spec results.
+ */
+const getSpecResults = (directory) => {
+  const specResults = fs.readdirSync(directory).filter((file) => (
+    file.indexOf('wdio-spec-results') > -1
+  ));
+
+  return specResults.map((file) => JSON.parse(fs.readFileSync(path.resolve(directory, file))));
+};
+
+/**
+ * Creates a unique filename for a test run based on the provided options.
+ * @param {Object} options - Test run options.
+ * @param {Object} options.browser - The browser name.
+ * @param {Object} options.formFactor - The test run form factor.
+ * @param {Object} options.locale - The test run locale.
+ * @param {Object} options.packageName - The package name.
+ * @param {Object} options.theme - The test run theme.
+ */
+const getResultsFileName = (options) => {
+  const {
+    browser,
+    formFactor,
+    locale,
+    packageName,
+    theme,
+  } = options;
+
+  const keys = [locale, theme, formFactor, browser, packageName].filter(key => !!key);
+
+  return `results-${keys.join('-')}.json`;
```

Comment:
since this doesn't need to be human readable, you could hash the keys string: https://nodejs.org/docs/v0.6.18/api/crypto.html#crypto_crypto_createhash_algorithm

or just use a uuid

Reply:
Should we make the 'results' key more unique to this reporter?

Reply:
I don't have a strong opinion. The current naming convention is aligning with the WDIO 4 spec reporter. I don't think the file names matter too much other than someone looking through the files manually (which might be the case on Jenkins artifacts). I'll defer to @ryanthemanuel 

If we do use a UUID we will need to clear the entire output directory before starting the test run to ensure all stale reports are cleared

Reply:
I'm thinking `terra-spec-reporter-<uuid>`. so you have a key you can use to clear, it won't conflict with other possible future reporters and the name *may* be shorter than if you use the full text.

Reply:
When we run the reporter, we'll want to be able to "load all files" and merge them together.  I think we could do that if they're all in the same directory or something even with uuids. The thing to remember is the flow in jenkins will be: spawn off all of our different possible combinations of locales, form factors, etc. these will have their own files that then need to be archived off for use later and *cannot* overwrite each other. I then think it is ok for orion-release-site to read them all in and merge them together. There will need to be enough data contained in the file to make sure that we know exactly what the contents were testing though.